File: 3345-smallest-divisible-digit-product-i/3345-smallest-divisible-digit-product-i.py
class Solution:
    def smallestNumber(self, n: int, t: int) -> int:
        i=n
        while True:
            copy =i
            prod=1
            while(copy>0):
                digit = copy % 10
                prod *= digit
                copy //= 10
            if(prod % t == 0):
                return i
            i +=1

File: 3345-smallest-divisible-digit-product-i/3345-smallest-divisible-digit-product-i.cpp
class Solution {
public:
    int smallestNumber(int n, int t) {
        int i = n;
        while(true){
            int copy = i;
            int prod = 1;
            while( copy > 0){
                int digit = copy % 10;
                prod = prod * digit;
                copy = copy / 10;
            }
            if(prod % t == 0){
                return i;
            }
            i++;
        }
    }
};